go/consensus/tendermint/apps/roothash/roothash.go

```diff
@@ -358,15 +355,16 @@ func (app *rootHashApplication) ForeignExecuteTx(ctx *abci.Context, other abci.A
 				if bytes.Equal(pair.GetKey(), registryapp.KeyRuntimeRegistered) {
 					var rt registry.Runtime
 					if err := cbor.Unmarshal(pair.GetValue(), &rt); err != nil {
-						return errors.Wrap(err, "roothash: failed to deserialize new runtime")
+						return fmt.Errorf("roothash: failed to deserialize new runtime: %w", err)
 					}
 
 					ctx.Logger().Debug("ForeignDeliverTx: new runtime",
 						"runtime", rt.ID,
 					)
 
-					ensureGenesis()
```

Comment:
:information_source: turns out the genesis arg of onNewRuntime is only used in InitChian